ref: initial structure (#1)
ohbm/cobidas: README.md

# COBIDAS checklist

> This repository is a first step in trying to turn the report published by the Committee on Best Practices in Data Analysis and Sharing (COBIDAS) of OHBM into an easy to use checklist to improve methods and results reporting in neuroimaging.

Reviewer: ```suggestion
This repository is a first step in trying to turn the report published by the Committee on Best Practices in Data Analysis and Sharing (COBIDAS) of OHBM into a practical tool for improving methods and results reporting in neuroimaging.
```

> This should a place where:

Reviewer: ```suggestion
This repository should be a place where:
```

> 1. issues and ideas around reaching that goal are discussed,

Reviewer: ```suggestion
1. issues and ideas around reaching this goal are discussed;
```

> 2. plans on how to get to this goal are designed and implemented.

Reviewer: ```suggestion
2. plans for how to reach this goal may be designed and implemented.
```

## How to reach us
[The github repository of this project can be found here](https://github.com/Remi-Gau/COBIDAS_chckls)

> Come and join us at on the `cobidas_checklist` channel on the [![slack_brainhack_3](https://user-images.githubusercontent.com/6297454/47951457-5b37b780-df61-11e8-9d77-7b5a4c7af875.png)](https://brainhack-slack-invite.herokuapp.com/).

Reviewer: ```suggestion
Come and join us on the `cobidas_checklist` channel at [![slack_brainhack_3](https://user-images.githubusercontent.com/6297454/47951457-5b37b780-df61-11e8-9d77-7b5a4c7af875.png)](https://brainhack-slack-invite.herokuapp.com/).
```

> Otherwise you can open a new issue if there is something you would like to discuss directly here.

Reviewer: ```suggestion
Otherwise you can open a new issue on the repository itself if there is something you would like to discuss directly here.
```

## Project Description
In 2012, in a [review of the methods and results reporting of more than 200 fMRI paper](https://www.ncbi.nlm.nih.gov/pubmed/22796459) the author found that "_Although many journals urge authors to describe their methods to a level of detail such that independent investigators can fully reproduce their efforts, the results described here suggest that few studies meet this criterion._"

> A few years ago, in order to improve replicability in f/MRI research, the Committee on Best Practices in Data Analysis and Sharing ([COBIDAS](https://www.humanbrainmapping.org/i4a/pages/index.cfm?pageid=3728)) of OHBM released a [report](https://www.biorxiv.org/content/10.1101/054262v2) to promote best practices for methods and results reporting. This was recently followed by a [similar initiative for EEG and MEG](https://osf.io/a8dhx/).

Reviewer: ```suggestion
In order to address this issue of replicability in f/MRI research, [COBIDAS](https://www.humanbrainmapping.org/i4a/pages/index.cfm?pageid=3728) released a [guidelines](https://www.biorxiv.org/content/10.1101/054262v2) for best practices in reporting of methods and results. This was soon followed by a [similar initiative for EEG and MEG](https://osf.io/a8dhx/).
```

Contrary to what the most optimistic people might have thought, these guidelines do not seem to have been widely adopted and anecdotal evidence ([see that twitter poll and thread](https://treeverse.app/view/Xf3jfvIZ)) suggests that, even among people who know about the report, few of them use it to write or review papers.

> One possible reason for this might have the unwieldy nature of the report. Anticipating on this issue, the authors of the report had included a checklist that still ended up taking almost 30 of the 70 pages of the whole document. Anyone who has used this checklist tends to agree that it is a great resource but that it is a bit cumbersome to use.

Reviewer: ```suggestion
One possible reason for this might be the unwieldy nature of the report. Anticipating this issue, the authors of the guidelines included a checklist (Appendix D) that still ended up taking almost 30 of the 70 pages of the whole document. Anyone who has used this checklist tends to agree that it is a great resource but that it is a bit cumbersome to interpret and apply.
```

### Short term goal

> **The short term goal of this project is to make this checklist easier to use**.

Reviewer: ```suggestion
**The short term goal of this project is to make the Appendix D checklist easier to use**.
```

> One way would be to turn it into a website that people can simply click their way through.

Reviewer: ```suggestion
One way to achieve this may be to turn the checklist into a website that users can click through and provide the information requested by COBIDAS.
```

> Ideally this would generate a small text file (for example a json file) that summarize what option was chosen for each item of the checklist.

Reviewer: ```suggestion
Ideally this would generate a small text file (for example a json file) that summarizes the response provided for each item of the checklist.
```

This machine readable file could then be used to automatically generate part of the methods section of an article.

#### Extensions

> Filling in that checklist could still be long and tedious so a way to facilitate this further would be read as much metadata as possible from existing standards like:

Reviewer: ```suggestion
The process of filling in an online checklist could be simplified if available metadata was drawn in from machine readable files created in the fulfilment of existing standards, for example:
```

1. the [Brain imaging data structure](http://bids.neuroimaging.io/)  used for that study (for example with respect to the data acquisition and experimental design).
2. the [NIDM results](http://nidm.nidash.org/specs/nidm-results_130.html) of any mass-univariate analysis performed for this study.

> 

Reviewer: Not sure I understand this bit. Are you suggesting reading in the BIDS participants or sequence json files for TR, TE etc.? Perhaps you could give some more explicit examples? I'm also not familiar with NIDM, so examples might be useful here too.

Reviewer 2: Examples are very good idea. Will add some.

#### Further developments

> Other off-shoot of this project could include developing toolboxes / packages / plug-ins for the main neuroimaging software (SPM, FSL, AFNI, Freesurfer) that would take some processing batch file as input (e.g matlabbatch.mat for SPM) and would output json files that could be used on the COBIDAS checklist website to automatically generate methods section (in a similar way to [what fMRIprep does](https://fmriprep.readthedocs.io/en/stable/citing.html))

Reviewer: ```suggestion
Other off-shoot of this project could include developing toolboxes / packages / plug-ins for the main neuroimaging software (SPM, FSL, AFNI, Freesurfer) that would take a processing batch file as input (e.g matlabbatch.mat for SPM or design.fsf for FSL) and output json files that could be used on the proposed COBIDAS checklist website to automatically generate methods section (in a similar way to [fMRIprep](https://fmriprep.readthedocs.io/en/stable/citing.html))
```

> Other off-shoot of this project could include developing toolboxes / packages / plug-ins for the main neuroimaging software (SPM, FSL, AFNI, Freesurfer) that would take some processing batch file as input (e.g matlabbatch.mat for SPM) and would output json files that could be used on the COBIDAS checklist website to automatically generate methods section (in a similar way to [what fMRIprep does](https://fmriprep.readthedocs.io/en/stable/citing.html))

Reviewer: Do you want to include fmriprep as a "main neuroimaging software"? Feels like it would be a relevant application (I know I'm going to be scoring the fmriprep documentation to find details for COBIDAS!)

Reviewer 2: Yeah I am not sure that fMRIprep would fall in that category because it bundles together so many other softwares.
But it would make sense to have an option "data preprocessed with fMRIprep" (and eventually point to a json file because fMRIprep still has options) and it would automatically fill in the corresponding part of the checklist.

Actually that makes me think that having the same possibility for other pre-processing pipelines that exist out there might be a good idea. 

Note to self: think about possibilities to link this more generally with nipype, and the PORCUPINE and GIRAFFE pipeline designing tools that Tim van Mourik has been working on.

Reviewer 2: I will rephrase and extend that.
